File: javascript/hamming/hamming.js
var compute = function(strand1, strand2) {
    if (strand1.length !== strand2.length) {
        throw new Error('DNA strands must be of equal length.');
    }
    
    if (strand1 === strand2) {
        return 0;
    }

    var hammingDistance = 0;
    var strandsLength = strand1.length;
    
    for (var i = 0; i < strandsLength; ++i) {
        if (strand1[i] !== strand2[i]) {
            ++hammingDistance;
        }
    }

    return hammingDistance;
};

module.exports = compute;

File: python/hamming/hamming.py
# class Hamming:
def distance(strand1, strand2):
	if len(strand1) != len(strand2):
		return 0
	
	if strand1 == strand2:
		return 0
	
	hammingDistance = 0
	
	strandsLength = len(strand1)
	for i in range(0, strandsLength):
		if strand1[i] != strand2[i]:
			hammingDistance += 1
	
	return hammingDistance